dotnet/Simple.Sauce/SauceOptions.cs

```diff
@@ -9,15 +11,31 @@ public class SauceOptions
     {
         private const string DefaultBrowserVersion = "latest";
         private const string DefaultPlatform = "Windows 10";
+        private readonly string _sauceUserName = Environment.GetEnvironmentVariable("SAUCE_USERNAME");
+        private readonly string _sauceAccessKey = Environment.GetEnvironmentVariable("SAUCE_ACCESS_KEY");
+        private readonly Dictionary<string, object> _sauceConfiguration;
 
-        public SauceOptions()
+        public SauceOptions() : this(new DriverFactory())
         {
+        }
+
+        public SauceOptions(IDriverFactory driverFactory)
+        {
+            DriverFactory = driverFactory;
+            _sauceConfiguration = new Dictionary<string, object>
+            {
+                ["username"] = _sauceUserName,
+                ["accessKey"] = _sauceAccessKey
+            };
             WithChrome();
         }
 
+        public IDriverFactory DriverFactory { get; }
         public EdgeOptions ConfiguredEdgeOptions { get; set; }
```

Comment:
we shouldn't need to have separate configurations for each browser implementation. They are all type `DriverOptions`, we just need to be able to get the formatting to the Session when asked.